### PR TITLE
Adding Detector metric for running any classifier from huggingface as a metric

### PR DESCRIPTION
**Pull request by mnagired**
This PR is a simple addition to the `metrics` file in order to support incoming risk detectors. 

Detectors are small, typically BERT-like, classification models which take as input a sentence and output a probability score.

The current test file is a placeholder, using an [open source model](https://huggingface.co/MilaNLProc/bert-base-uncased-ear-misogyny) while we work through clearances for the detectors.

### CODE REVIEW

File: tests/library/test_metrics.py [hunk 1]
```diff
@@ -698,7 +699,19 @@ def test_kendalltau(self):
         outputs = apply_metric(
             metric=metric, predictions=predictions, references=references
         )
-        global_target = 0.81649658092772
+        global_target = 0.6025716662406921
+        self.assertAlmostEqual(global_target, outputs[0]["score"]["global"]["score"])
+
```
Comment (elronbandel) [inline tests/library/test_metrics.py:704]:
```suggestion
        global_target = 0.81649658092772
```
Seems like you override kandall tau test by accident

File: tests/library/test_metrics.py [hunk 2]
```diff
@@ -698,7 +699,19 @@ def test_kendalltau(self):
...
+    def test_detector(self):
+        metric = Detector(model_name = "MilaNLProc/bert-base-uncased-ear-misogyny")
+        predictions = ["I hate women.", "I do not hate women."]
+        references = [["I hate women."], ["I do not hate women."]]
+        outputs = apply_metric(
+            metric=metric, predictions=predictions, references=references
+        )
+        
+        global_target = 0.9562818706035614
         self.assertAlmostEqual(global_target, outputs[0]["score"]["global"]["score"])
```
Comment (elronbandel) [inline tests/library/test_metrics.py:715]: You got here this error:
```
    self.assertAlmostEqual(global_target, outputs[0]["score"]["global"]["score"])
AssertionError: 0.9562818706035614 != 0.9562810361385345 within 7 places (8.344650268554688e-07 difference)
```
I think an appropirate fix will be to check within 4 decimal points: 
```suggestion
        self.assertAlmostEqual(global_target, outputs[0]["score"]["global"]["score"], places=4)
```

File: tests/library/test_metrics.py (hunk 1 above)
Comment (mnagired) [reply tests/library/test_metrics.py]: you're right. apologies, I added this back in the latest commit! thanks for catching

File: tests/library/test_metrics.py (hunk 2 above)
Comment (mnagired) [reply tests/library/test_metrics.py]: thank you - also fixed in another commit!

**Review by elronbandel — CHANGES_REQUESTED** [review]
Well done this looks really good! This will defentily be useful for many others, so thank you!

I added few comments on speficic code lines, will be great if you can go over them. 

In order to pass all the tests you will need to run the pre-commits hooks:
1. First install the dev requiremnts `pip install requirements/dev.rqr`
2. Then run `make pre-commit`. Some issues will be fixed automatically, some you will might need to solve. Then commit the changes back and you will pass the `Code Quality and Security Tests`

**Comment by mnagired** [comment]
Thanks @elronbandel !

- I addressed the comments about the overriding of test_kendalltau and the decimal rounding issue
- I've installed the dev requirements and committed the latest version which passes the pre-commit hooks with no issues

Looks like the Code Quality check has passed! Is there anything else to do?

**Review by elronbandel — APPROVED** [review]
Looks great! Well done.

**Comment by elronbandel** [comment]
@mnagired notice, that this metric is not yet in the unitxt catalog. 
In order to add a specific metric to the unitxt catalot you will need to write a file like this one:
https://github.com/IBM/unitxt/blob/main/prepare/metrics/kendalltau.py
And run it and get the resulting json and add it ass well. 
If you wish you can add to the catalog a new metric called:
`metrics.hate.<model_name>` or any otther name as you find suitable. 
This will add the metric to the catalog here: https://unitxt.readthedocs.io/en/latest/catalog.metrics.html
This will also allow pepole to use your metric with the specific model you choose in production.

**Comment by mnagired** [comment]
Thank you for the pointer!

Yes, we will add the metric files for each detector individually as each model gets approved so that they can be added to the catalog.

**Comment by mnagired** [comment]
looks like all checks have passed! please merge whenever you get a chance to. 

thanks a lot @elronbandel for your help!!

**Comment by elronbandel** [comment]
Your welcome! It's merged🎉

**Pull request merged**